src/averageAges.js

```diff
@@ -1,64 +1,50 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+const FEMALE = 'f';
+const MALE = 'm';
+const HUNDRED = 100;
+
+function getAverageAge(people) {
+  return people.reduce((accum, { died, born }) =>
+    (accum + (died - born)), 0) / people.length;
+};
+
+function getAgeDiff(people1, people2) {
+  return ((people1.reduce((accum, { born, mother }) =>
+    accum + born - people2.find(({ name }) => mother
+      === name).born, 0)) / people1.length).toFixed(2);
+};
```

Comment:
Now everything is written correctly, but such code is difficult to read and maintain in the future. Let's make it a little simpler:
```suggestion
// we should name this function with "...Average..." because we return not different but literally average difference
function getAverageAgeDiff(childrens, people) { // we can name first arg "childrens" because "sons" also childrens
  const ageDiff = childrens.reduce((acc, { born, mother: motherName }) => {
    const motherBirthYear = people.find(({ name }) => motherName === name).born; // better to find this value separately is to improve code readability.
    return acc + born - motherBirthYear;
  }, 0);
  return +(ageDiff / childrens.length).toFixed(2); // if you move the `+` here, it will not be necessary to write it during the call and it will look better
};
```
I wrote the comments exclusively for you, you don't need to leave them in the code.